### PR TITLE
IP Validation done

### PR DESCRIPTION
**Pull request by MNBuyskih**
IP Validation (2p)
https://www.codewars.com/kata/515decfd9dcfc23bb6000006

### CODE REVIEW

File: ip_validation/ip_validation.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require 'ipaddr'
+
+def ip_validation(str)
```
Comment (noku) [inline ip_validation/ip_validation.rb]: 1. in ruby we could name the methods with `?` at the end to signify that this method returns a boolean.
  `ip_validation(str)` → `ipv4?(str)`
2. we could drop `begin` inside the method and just apply `rescue` to the block.

so the final solution could look like this: 

```ruby
def ipv4?(str)
  IPAddr.new(str).ipv4?
rescue IPAddr::InvalidAddressError
  false
end
```

Comment (MNBuyskih) [reply ip_validation/ip_validation.rb]: Refactored

File: ip_validation/ip_validation_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'rspec'
+require_relative 'ip_validation'
+
+describe '#ipv4?' do
+  describe 'for valid addresses' do
```
Comment (noku) [inline ip_validation/ip_validation_spec.rb]: `nitpick` probably in this case is good to use `context`. [betterspecs](http://www.betterspecs.org/#contexts), 

(same with for `for invalid addresses`)

File: ip_validation/ip_validation_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    it 'does returns true' do
```
Comment (noku) [inline ip_validation/ip_validation_spec.rb]: `nitpick` we can just write `it 'returns true' do`

Comment (MNBuyskih) [reply ip_validation/ip_validation_spec.rb]: Done

File: ip_validation/ip_validation_spec.rb (hunk 2 above)
Comment (MNBuyskih) [reply ip_validation/ip_validation_spec.rb]: Done

**Review by noku — APPROVED** [review]

**Pull request closed without merging**